autopush/websocket.py

```diff
@@ -219,6 +241,11 @@ def process_notifications(self, tries=0):
         if self._should_stop:
             return
 
+        # Are we paused? Try again later.
+        if self.paused:
+            reactor.callLater(1, self.process_notifications)
```

Comment:
It's possible that process_notifications was already scheduled, so that wouldn't reduce the need for this check.